tasks/main.yml

```diff
@@ -14,3 +19,20 @@
   tags:
     - consul_keystore_configure
   when: consul_keystore_name is defined
+
+- name: Consul | Delete diff keys
+  consul_kv:
+    host: "{{ consul_keystore_server }}"
+    port: "{{ consul_keystore_http_port }}"
+    token: "{{ consul_keystore_acl_token }}"
+    key: "{{ key }}"
+    state: absent
+  loop: "{{ consul_backup_all_properties | difference(consul_set_properties) }}"
+  loop_control:
+    loop_var: key
+  when:
+    - consul_keystore_name is defined
+    - consul_backup_all_properties is defined
```

Comment:
With the set_fact running on line 4, wouldn't the properties be always defined?

Reply:
Mm, yes I guess. I can remove the is defined checks.